### PR TITLE
Async Conversion

### PR DESCRIPTION
**Pull request by JoschuaL**
Converts all applicable functions of the library to use async web requests.

I tried to keep the API as-is as much as I could, so aside from needing async/await now, the only bigger change 
ended up being ListIterator not implementing IntoIter anymore (as async iterators dont exist in the same way yet)
It instead has a next_stream and into_stream function.
next_stream can be used the same as an async version of next(), where as into_stream can be used to convert the ListIter into a futures::stream::Stream, which is as close to async iterators one can currently get afaik.
Most functions that apply to iters apply to streams using futures::stream::StreamExt (map, filter, collect etc.)
The only bigger omission i found while porting the doctests was find, which to be fair, probably should use a single card query in the first place.

Of course I'm open to make any changes neccessary. I would also understand if the maintainer(s) prefer to keep to synchronous api, in which case i will probably maintain an async fork (which I would then ask to be allowed to upload to crates.io with an appropriate name like scryfall-rs-async or something like that) as I do need async capabilities for a project im currently woking on.

### CODE REVIEW

File: src/card/frame_effect.rs [hunk 1]
```diff
@@ -68,6 +68,8 @@ pub enum FrameEffect {
     Textless,
     /// Is a story spotlight card
     StorySpotlight,
+    /// Undocumented
+    Gilded,
```
Comment (mendess) [inline src/card/frame_effect.rs:72]: Is this undocumented in scryfall's webiste?

File: src/card.rs [hunk 2]
```diff
@@ -211,7 +211,7 @@ pub struct Card {
     pub toughness: Option<String>,
 
     /// The type line of this card.
-    pub type_line: String,
+    pub type_line: Option<String>,
```
Comment (mendess) [inline src/card.rs:214]: why did you change these to option? Is the because of the weird double faced cards from the "heads I win, tails you lose" cards?

Comment (JoschuaL) [reply src/card.rs:214]: the parsing of the bulk files containing all cards/one print per card etc. failed because some cards are in fact missing/dont have a type line. I forgot which ones were the problem, but I think it was something from an un set

File: src/card/frame_effect.rs (hunk 1 above)
Comment (JoschuaL) [reply src/card/frame_effect.rs:72]: at least i couldnt find anything about it in their api docs

**Comment by JoschuaL** [comment]
after digging into some execution graphs, i noticed that the stream from ListIter was compleated done sequentially, so I also added the option to buffer it for some simple parallelism.

File: src/card.rs (hunk 2 above)
Comment (JoschuaL) [reply src/card.rs:214]: never mind, it’s probably because of heads & tails, the first card that threw an error was kraks thumb

File: src/card/frame_effect.rs (hunk 1 above)
Comment (JoschuaL) [reply src/card/frame_effect.rs:72]: thinking about it, it might be [this](https://secretlair.wizards.com/eu/de/product/724663/showcase-streets-of-new-capenna-gilded-foil-edition)

File: src/card.rs (hunk 2 above)
Comment (mendess) [reply src/card.rs:214]: Yeah, I have never been able to find a solution for those cards I like, every card has those fields except for the 3 double faced cards in that deck, so I really wanted to avoid the unergonomic change but I haven't been able to find a better solution yet so maybe I should surrender to the options?

If so, can you document that these new option fields are present in every card except the 3 double sided ones (and possibly link the cards in the docs)

**Comment by mendess** [comment]
Thanks for this PR and I'm sorry it's taking me so long to get back to you, work was being really time-consuming. 

I think overall the PR is good, I'm just annoyed at krark's thumb 😂

**Comment by JoschuaL** [comment]
> 

no peoblem, expected as much (the christmas-new year timing probably didnt help it either ;P)
Im also a bit irritated by cards not having type information for some reason, so I just hope that scryfall maybe changes that on their side at some point.

**Comment by mendess** [comment]
@JoschuaL I think I'm gonna merge. Can you just bump the version to `0.12.0`, please?

**Comment by JoschuaL** [comment]
assuming changing the version in Cargo.toml was all that was needed that should be done now. Happy this was useful.

**Review by mendess — APPROVED** [review]
